app/decorators/models/solidus_kustomer/spree/user/kustomer_lifecycle.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module SolidusKustomer
+  module Spree
+    module User
+      module KustomerLifecycle
+        def self.prepended(base)
+          base.after_commit :create_customer_on_creation, on: :create
+          base.after_commit :update_customer_on_update, on: :update
+        end
+
+        private
+
+        def create_customer_on_creation
+          return unless SolidusKustomer.configuration.create_customer_on_user_creation
+
+          SolidusKustomer.create_customer_later(self)
+        end
+
+        def update_customer_on_update
+          return unless SolidusKustomer.configuration.update_customer_on_user_update
+
```

Comment:
Do you think it also makes sense to check if any relevant attributes have changed? Otherwise, you'll be calling the Kustomer API every time a user is updated, even when nothing is supposed to change in Kustomer.